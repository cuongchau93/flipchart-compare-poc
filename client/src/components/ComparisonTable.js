import React, { Component } from 'react';
import { Table } from 'reactstrap';
import FeatureRow from './FeatureRow';
import '../styles/ComparisonTable.css';

class ComparisonTable extends Component {
    render() {
        const rows = [];

        if(this.props.features){
            if(this.props.features.Highlights){
                Object.keys(this.props.features.Highlights).forEach((spec)=>{
                    rows.push(
                        <FeatureRow
                            className="highlighted"
                            name={spec}
                            feature1={this.props.features.Highlights[spec][0]}
                            feature2={this.props.features.Highlights[spec][1]}
                            key={spec+this.props.features.Highlights[spec][0]+this.props.features.Highlights[spec][1]} />
                    );
                });
            }

            if(this.props.features.Specifications){
                Object.keys(this.props.features.Specifications).forEach((spec)=>{
                    rows.push(
                        <FeatureRow
                            className="normal"
                            name={spec}
                            feature1={this.props.features.Specifications[spec][0]}
                            feature2={this.props.features.Specifications[spec][1]}
                            key={spec+this.props.features.Specifications[spec][0]+this.props.features.Specifications[spec][1]} />
                    );
                });
            }
        }

        return (
            <Table responsive hover bordered>
                <thead>
                    <tr>
                        <th>Features</th>
                        <th>Product 1</th>
                        <th>Product 2</th>
                    </tr>
                </thead>
                <tbody>{rows}</tbody>
            </Table>
        );
    }
}

export default ComparisonTable;