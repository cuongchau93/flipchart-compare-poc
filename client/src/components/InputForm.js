import React, { Component } from 'react';
import '../styles/InputForm.css';
import { Button, Form, FormGroup, Label, Input} from 'reactstrap';

class InputForm extends Component {
    constructor(props) {
        super(props);

        this.handleSubmit = this.handleSubmit.bind(this);
        this.state = {
            loading: false
        }
    }

    handleSubmit(event) {
        if(event.target.url1.value && event.target.url2.value){
            this.props.onSubmission(event.target.url1.value ,event.target.url2.value);
        }
        event.preventDefault();
    }

    render() {
        return (
            <Form onSubmit={this.handleSubmit} id="myForm">
                <FormGroup>
                    <Label for="url1">First URL</Label>
                    <Input type="url" id="url1" name="url1" defaultValue="https://www.lazada.sg/products/samsung-galaxy-s8-64gb-midnight-black-i106217842-s108237124.html" />
                </FormGroup>
                <FormGroup>
                    <Label for="url2">Second URL</Label>
                    <Input type="url" id="url2" name="url2" defaultValue="https://www.lazada.sg/apple-iphone-8-256gb-2gb-ram-grey-60291398.html" />
                </FormGroup>
                <Button>Submit</Button>
            </Form>
        );
    }
}
export default InputForm;