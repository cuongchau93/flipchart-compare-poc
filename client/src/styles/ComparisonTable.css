thead {
  background-color: #b9c9fe; }

.highlighted {
  background-color: #e8edff; }
