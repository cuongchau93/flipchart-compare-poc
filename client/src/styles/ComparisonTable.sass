$lightPurple: #b9c9fe
$lighterPurple: rgb(232, 237, 255)

thead
    background-color: $lightPurple

.highlighted
    background-color: $lighterPurple
