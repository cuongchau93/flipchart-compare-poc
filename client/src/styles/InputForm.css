#myForm {
  background-color: lightgray;
  padding: 20px; }

#myForm input {
  border: 1px solid lightgray;
  padding: 12px 20px;
  border-radius: 4px;
  box-sizing: border-box; }
