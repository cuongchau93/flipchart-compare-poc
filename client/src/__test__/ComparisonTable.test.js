import React from 'react';
import InputForm from '../components/InputForm';
import { shallow,configure } from 'enzyme'
import renderer from 'react-test-renderer';
import ComparisonTable from '../components/ComparisonTable';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

const sampleSpec = {
    "Specifications": {
        "RAM": "32G"
    },
    "Highlights": {
        "Camera": "16MP"
    }
};

describe("ComparisonTable " , () => {
    const component = shallow(
        <ComparisonTable features={sampleSpec}/>
    );

    it('renders and matches our snapshot', () => {
        const component = renderer.create(
            <ComparisonTable />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});