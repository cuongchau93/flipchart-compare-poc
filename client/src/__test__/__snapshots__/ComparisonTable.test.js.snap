// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonTable  renders and matches our snapshot 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Features
        </th>
        <th>
          Product 1
        </th>
        <th>
          Product 2
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
